src/ceph_fuse.cc

```diff
@@ -130,6 +130,15 @@ int main(int argc, const char **argv, const char *envp[]) {
   // args for fuse
   const char **newargv;
   int newargc;
+
+  // add the -d options back which remove by gloabl_init()
+  for (int i = 0; i < argc; i++) {
+    if (!strcmp("-d", argv[i])) {
+      args.insert(args.end(), "-d");
+      break;
+    }
+  }
+
```

Comment:
doing this manual checking is not quite right. It'd be better to detect this arg in the above for loop?

Reply:
>  It'd be better to detect this arg in the above for loop?
@batrick  I didn't get this, could you explain more ?
Thanks

Reply:
you have a `for` loop iterating through `args`, i wonder what Patrick wants is to detect "-d" there as well.

Reply:
Okay, but this won't work, because the line#97 have already deleted the `-d` from the `args` vector.

Reply:
I believe you may be right that the `-d` option is stripped from `args` but i don't see where. HEre's where it is processed:

https://github.com/ceph/ceph/blob/617298c201f8bbb9c16ff0a43c13d6dd93f90f82/src/common/config.cc#L672-L678

Looks to me it should stay in the args vector. In any case, we could add a config option to turn on fuse debugging and set that config option in the above code processing the `-d` flag. That would be better in any case I think as then we could turn on FUSE debugging without using the `-d` flag at all.

Reply:
> I believe you may be right that the `-d` option is stripped from `args` but i don't see where. HEre's where it is processed:
> https://github.com/ceph/ceph/blob/617298c201f8bbb9c16ff0a43c13d6dd93f90f82/src/common/config.cc#L672-L678

Yeah, it is here:
`global_init() --> global_pre_init() --> conf.parse_argv() --> ceph_argparse_flag(args, i, "-d", (char*)NULL)`

And the `ceph_argparse_flag(args, i, "-d", (char*)NULL)` will erase it.


> 
> Looks to me it should stay in the args vector. In any case, we could add a config option to turn on fuse debugging and set that config option in the above code processing the `-d` flag. That would be better in any case I think as then we could turn on FUSE debugging without using the `-d` flag at all.

The following is the usage info for the `ceph-fuse` command, and the `FUSE options:` segment was not under ceph's control, it is printed by the libfuse's usage helper.

I am thinking when people check the usage from `ceph-fuse -h` output, and in `FUSE options` segment the `-d` won't work as expected, won't it be confusing ?

```
[root@lxbceph0 build]# ./bin/ceph-fuse -h
usage: ceph-fuse [-n client.username] [-m mon-ip-addr:mon-port] <mount point> [OPTIONS]
  --client_mountpoint/-r <sub_directory>
                    use sub_directory as the mounted root, rather than the full Ceph tree.

usage: ceph-fuse [options] <mountpoint>

FUSE options:
    -h   --help            print help
    -V   --version         print version
    -d   -o debug          enable debug output (implies -f)
    -f                     foreground operation
    -s                     disable multi-threaded operation
    -o clone_fd            use separate fuse device fd for each thread
                           (may improve performance)
    -o max_idle_threads    the maximum number of idle worker threads
                           allowed (default: 10)
    -o allow_other         allow access by all users
    -o allow_root          allow access by root
    -o auto_unmount        auto unmount on process termination

  --conf/-c FILE    read configuration from the given configuration file
  --id ID           set ID portion of my name
  --name/-n TYPE.ID set name
  --cluster NAME    set cluster name (default: ceph)
  --setuser USER    set uid to user or uid (and gid to user's gid)
  --setgroup GROUP  set gid to group or gid
  --version         show version and quit
```

Reply:
The `daemon_debug` option may still not very graceful IMO, it will only make sense when the `-d` option is specified, and then we must do:

```
133   auto daemonize = g_conf().get_val<bool>("daemonize");
134   auto debug = g_conf().get_val<bool>("daemon_debug");                                                                                 
135   if (!daemonize && daemon_debug) {          
136       args.insert(args.end(), "-o");      
137       args.insert(args.end(), "debug");   
138   }

```
Or the simplest way is do not erase the `-d` option from the `args` vector when parsing in `common/config.cc` ?

Reply:
I have update the commit and please check whether is the right approach to fix it ?
Thanks.